packages/components/src/components/form/Range/Range.tsx

```diff
@@ -66,57 +60,85 @@ const Label = styled.div<{ $width?: number }>`
     font-size: ${variables.FONT_SIZE.TINY};
     font-weight: ${variables.FONT_WEIGHT.MEDIUM};
     opacity: 0.5;
-    cursor: pointer;
+    cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
+
+    :first-child {
+        text-align: left;
+    }
+`;
+
+const LabelsWrapper = styled.div<{ count: number; $width?: number }>`
+    display: grid;
+    grid-template-columns: ${({ count, $width }) =>
+        `repeat(${count}, ${$width ? `${$width}px` : '1fr'})`};
+    justify-content: space-between;
 `;
 
 export interface RangeProps {
     className?: string;
+    disabled?: boolean;
+    labels?: Array<{ value: string | number; component?: React.ReactNode }>;
     max?: number;
     min?: number;
-    step?: string;
     onChange: React.ChangeEventHandler<HTMLInputElement>;
+    onKeyDown?: React.KeyboardEventHandler;
```

Comment:
I changed the styles a bit so that the `box-shadow` is more distinctive and works in dark mode as well. I also switched the selector from `active` on the thumb to `focus` on the track which better corresponds to the editable state. I'd rather not blur `onMouseUp` to be consistent with default browser behaviour. 2b18b33f0d649b32bfba5fcdc521aacc007d3074

Reply:
Ok, let's not blur on mouse up! Couple things:


Please change the shadow to that, so it's more subtle and in line with the Switch behavior. 
```
 box-shadow: 0 0 3px 3px ${({ theme }) => theme.BOX_SHADOW_RANGE};
```

Also make it appear on `:focus-visible`, so it doesn't appear on grab and add `box-shadow: 0 0 1px 1px ${({ theme }) => theme.BOX_SHADOW_RANGE};` to the `:active` style. 🙏  That way thumbs in `Switch` and `Range` behave the same way.